js/controller.js

```diff
@@ -830,12 +830,16 @@ OO.plugin('Html5Skin', function(OO, _, $, W) {
 
     onVcPlay: function(event, source) {
       this.state.currentVideoId = source;
-      if (this.state.adWasPaused && this.state.currentAdsInfo && this.state.currentAdsInfo.currentAdItem && this.state.currentAdsInfo.currentAdItem.ssai) {
-        this.state.adPauseDuration = Date.now() - this.state.adPausedTime;
-        //we calculate new ad end time, based on the time that the ad was paused.
-        this.state.adEndTime = this.state.adEndTime + this.state.adPauseDuration; //milliseconds
-        this.state.adWasPaused = false;
-        this.state.adPauseDuration = 0;
+      if (this.state.currentAdsInfo.currentAdItem.ssai) {
```

Comment:
@ipadilla4 we should not check `this.state.currentAdsInfo.currentAdItem.ssai` before executing the null checks for `this.state.currentAdsInfo && this.state.currentAdsInfo.currentAdItem` that are inside the nested if. I would put `this.state.currentAdsInfo && this.state.currentAdsInfo.currentAdItem` as the top level check, with individual checks for `this.state.currentAdsInfo.currentAdItem.ssai` (during which we set currentVideoId to OO.VIDEO.MAIN) and `this.state.adWasPaused` (where we handle the pause logic) inside.